src/renderer.ts

```diff
@@ -156,6 +158,632 @@ export class DefaultCellRenderer implements ICellRendererFactory {
 /**
  * a renderer rendering a bar for numerical columns
  */
+
+class HeatmapCellRenderer extends DefaultCellRenderer {
+
+  createSVG(col: Column, context: IDOMRenderContext): ISVGCellRenderer {
+    function cell_dim(total, cells) {
+      return (total / cells);
+    }
+
+    const total_width = col.getWidth();
+    const bins = (<any>col.desc).datalength;
+    const defaultdomain = [0, 100];
+    var domain = (<any> col.desc).domain;
+    domain = (domain === undefined || null) ? defaultdomain : domain;
+    const min = domain[0], max = domain[1];
+    var colorrange = (<any> col.desc).colorrange;
+    const defaultcolor = ['blue', 'red'];
+    colorrange = (colorrange === undefined || null) ? defaultcolor : colorrange;
+    const celldimension = cell_dim(total_width, bins);
+    var color: any = d3.scale.linear<number, string>();
+    color = (min < 0) ? color.domain([min, 0, max]).range([colorrange[0], 'white', colorrange[1]])
+      : color.domain([min, max]).range(['white', colorrange[1]]);
+    const padding = context.option('rowPadding', 1);
+    return {
+
+      template: `<g class="heatmapcell"></g>`,
+      update: (n: SVGGElement, d: IDataRow, i: number) => {
+        const g = d3.select(n);
+        const rect = g.selectAll('rect').data(col.getValue(d.v, i));
+        rect.enter().append('rect');
+        rect
+          .attr({
+            y: padding,
+            x: (d, i) => (i === null || 0) ? 0 : (i * celldimension),
```

Comment:
`i` will always be the index and can't be null, and `0 * celldimension = 0` no need for the if